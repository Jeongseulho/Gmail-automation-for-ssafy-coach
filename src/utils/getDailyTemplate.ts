import { formatDate } from './formatDate';

export const getDailyTemplate = (date: string, campus: string, classSection: string, name: string) => {
  const formattedDate = formatDate(date);
  const title = '[SSAFY]' + ' ' + formattedDate + ' ' + campus + ' ' + classSection + ' ' + '실습코치' + ' ' + name + ' ' + '일일 보고서 송부';
  const content = `안녕하세요.
      <br>
      ${campus} 실습코치 ${name}입니다.
      <br>
      <br>
      ${formattedDate.slice(0, 5)} 일일 프로젝트 진행 보고서 첨부파일로 송부 드립니다.
      <br>
      <br>
      <b>특이사항</b>
      <br>
      해당 사항 없음.
      <br>
      <br>
      감사합니다.
      <br>
      이희진 드림.`;
  return {
    title,
    content,
  };
};
