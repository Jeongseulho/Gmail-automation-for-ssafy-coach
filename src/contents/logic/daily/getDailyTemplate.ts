import { Campus } from '@/types/Campus';
import { formatDate } from '@/utils/formatDate';

export const getDailyTemplate = (cohort: string, date: string, campus: Campus, classSection: string, name: string) => {
  const title = `[SSAFY] ${formatDate(date)} ${campus} ${classSection} 실습코치 ${name} 일일 보고서 송부`;
  const content = `
    <div>안녕하십니까, ${cohort} ${campus} 실습코치 ${name}입니다.</div>
    <br>
    <div>${formatDate(date)} 일일 프로젝트 진행 보고서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>- 해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>${name} 드림.</div>
  `;
  return {
    title,
    content,
  };
};
