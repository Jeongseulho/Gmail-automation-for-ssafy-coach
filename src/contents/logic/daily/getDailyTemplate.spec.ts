import { getDailyTemplate } from './getDailyTemplate';

describe('getDailyTemplate 함수', () => {
  it('일일 보고서 템플릿을 반환한다', () => {
    const { title, content } = getDailyTemplate('240202', '구미', '1반', '이희진');
    expect(title).toMatchSnapshot();
    expect(content).toMatchSnapshot();
  });
});
