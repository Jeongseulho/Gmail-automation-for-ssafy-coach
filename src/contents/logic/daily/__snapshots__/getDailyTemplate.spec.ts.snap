// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDailyTemplate 함수 일일 보고서 템플릿을 반환한다 1`] = `"[SSAFY] 2월 2일(금) 구미 1반 실습코치 이희진 일일 보고서 송부"`;

exports[`getDailyTemplate 함수 일일 보고서 템플릿을 반환한다 2`] = `
"
    <div>안녕하세요.</div>
    <div>구미 실습코치 이희진입니다.</div>
    <br>
    <div>2월 2일 일일 프로젝트 진행 보고서 첨부파일로 송부 드립니다.</div>
    <br>
    <b>특이사항</b>
    <br>
    <div>해당 사항 없음.</div>
    <br>
    <div>감사합니다.</div>
    <div>이희진 드림.</div>
  "
`;
